src/session.c

```diff
@@ -835,6 +918,12 @@ int session_end(SESSION_HANDLE session, const char* condition_value, const char*
         {
             result = 0;
         }
+
+        // all link endpoints are destroyed when the session end happens
+        for (i = 0; i < session_instance->link_endpoint_count; i++)
+        {
+            
+        }
```

Comment:
Is something missing in the loop?

Reply:
Yep, this was missing code indeed, I added here the removal of all endpoints associated with the session.

---
In reply to: [188977809](https://github.com/Azure/azure-uamqp-c/pull/236#discussion_r188977809) [](ancestors = 188977809)